<launch>
    <!-- 子文件中声明参数（带默认值） -->
    <arg name="map_size_x" default="42.0"/>
    <arg name="map_size_y" default="30.0"/>
    <arg name="map_size_z" default=" 5.0"/>

    <arg name="odom_topic" default="visual_slam/odom" />

    <!-- 继承基础配置 -->
    <include file="$(find ego_planner)/launch/run_in_sim.launch">
        
        <arg name="drone_id" value="0"/>
         <arg name="uav_type" value="carpet"/>

        <arg name="init_x" value="-20.0"/>
        <arg name="init_y" value="-9.0"/>
         <arg name="init_z" value="0.1"/>

        <arg name="target_x" value="20.0"/>
        <arg name="target_y" value="9.0"/>
        <arg name="target_z" value="1"/>

        <arg name="map_size_x" value="$(arg map_size_x)"/>
        <arg name="map_size_y" value="$(arg map_size_y)"/>
        <arg name="map_size_z" value="$(arg map_size_z)"/>
        <arg name="odom_topic" value="$(arg odom_topic)"/>
    </include>

    <!-- 加载uav1专用节点 -->
    <node pkg="uav1_scan_planner" type="scan_planner_node" name="scan_planner_node" output="screen">
        <param name="scan_width" value="4"/>
        <param name="scan_length" value="4"/>
        <param name="lane_spacing" value="1"/>
        <param name="forward_offset" value="1"/>
    </node>
</launch>